DetermineSite.cmake

```diff
@@ -11,8 +11,19 @@ elseif (${BUILD_SITE} MATCHES "pfe" OR ${BUILD_SITE} MATCHES "r[0-9]*i[0-9]*n[0-
 elseif (EXISTS /ford1/share/gmao_SIteam AND EXISTS /ford1/local AND ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
    set (DETECTED_SITE "GMAO.desktop")
 else ()
-   set (DETECTED_SITE ${BUILD_SITE})
+   file(DOWNLOAD http://169.254.169.254/latest/meta-data/instance-id instance-id
```

Comment:
I'm guessing you don't care about the contents of the file or saving it to disk. Perhaps we should add a no-save option to `file(DOWNLOAD)`, or provide a variable specifying the bit bucket (`nul` on Windows, `/dev/null` everywhere else.)

Anyway, LGTM.

Reply:
@KyleFromKitware I suppose it might be nice to have a way to not safe the downloaded output. Maybe something for CMake 3.19 😄

Reply:
I've just implemented this. Please see https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5060